perception/euclidean_cluster/include/euclidean_cluster/euclidean_cluster.hpp

```diff
@@ -29,9 +29,8 @@ class EuclideanCluster : public EuclideanClusterInterface
   EuclideanCluster();
   EuclideanCluster(bool use_height, int min_cluster_size, int max_cluster_size);
   EuclideanCluster(bool use_height, int min_cluster_size, int max_cluster_size, float tolerance);
-  bool cluster(
-    const pcl::PointCloud<pcl::PointXYZ>::ConstPtr & pointcloud,
-    std::vector<pcl::PointCloud<pcl::PointXYZ>> & clusters) override;
+  std::shared_ptr<autoware_auto_perception_msgs::msg::DetectedObjects> cluster(
+    const pcl::PointCloud<pcl::PointXYZ>::ConstPtr & pointcloud) override;
```

Comment:
So, it appears to me that the reason that they used `std::vector<pcl::PointCloud<pcl::PointXYZ>> & clusters` instead of a ROS message is so that this package could avoid ROS dependencies for separation-of-concerns. I think this is the way it should continue to be. Did you have rationale for changing it?

Reply:
I have added another conversion function inside the utils.cpp and restored this back to original, addressed in https://github.com/upowerrobotics/autoware.universe/pull/4/commits/4c625fbeaad3c87b187ebb30d3caf125d75be5e5